Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragmentViewModel.kt

```diff
@@ -191,13 +195,28 @@ class HomeFragmentViewModel @AssistedInject constructor(
         localStatisticsProvider.current,
         networkStateProvider.networkState.map { it.isInternetAvailable }.distinctUntilChanged()
     ) { statsData, localStatsData, isInternetAvailable ->
-        statsData.copy(
-            items = mutableListOf(
+
+        val cardIdSequence = statsData.cardIdSequence
+        val ordering = Ordering.explicit(cardIdSequence.toList())
+        val stats = localStatsData.items.plus(statsData.items)
+            .filterIsInstance<StatsSequenceItem>()
+            .filter { it.cardType.id in cardIdSequence }
+            .sortedWith { a, b ->
+                ordering.compare(a.cardType.id, b.cardType.id)
+            }
+
+        val addStatsItems = when {
+            !cardIdSequence.contains(StatsType.LOCAL_INCIDENCE.id) -> emptySet<StatsItem>()
```

Comment:
I'm not sure if this is alright. I agree that it's probably better to not show the statistics at all if we don't get them from the server, but the current version will display a card that tells the user that statistics are not available due to no internet connection, or something along those lines. I think this should be clarified with UX.